sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestHelper.scala

```diff
@@ -78,9 +80,12 @@ trait SQLQueryTestHelper {
    * @param result a function that returns a pair of schema and output
    */
   protected def handleExceptions(result: => (String, Seq[String])): (String, Seq[String]) = {
+    val format = MINIMAL
     try {
       result
     } catch {
+      case e: SparkThrowable with Throwable if e.getErrorClass != null =>
+        (emptySchema, Seq(e.getClass.getName, getMessage(e, format)))
```

Comment:
We should normalize the error message as before, see https://github.com/apache/spark/pull/37452/files#diff-739e09a7f5e5fd55fb42ca80b075e0c45f4342853ba66b0df1631310e691950fR94

Reply:
@cloud-fan Do you have an example where we output a plan in an error with error class. As far as I know we are trying to avoid output any plans when we migrate on error classes.

Reply:
The sql fragment (query context) may contain expr IDs.